File: bench/list-ops.js
const n  = 100000;
const t0 = Date.now();

const xs = [];
for (let i = 0; i < n; i++) xs.push(i);
const result = xs.map(x => x * 2).filter(x => x % 3 === 0).reduce((a, x) => a + x, 0);

const t1 = Date.now();

console.log(`BENCH_MS: ${t1 - t0}`);
console.log(`result=${result}`);

File: tests/bench/list-ops.scala
//> using scala 3

@main def run(): Unit =
  val n  = 100000
  val t0 = System.nanoTime()

  val xs     = List.range(0, n)
  val result = xs.map(x => x * 2).filter(x => x % 3 == 0).foldLeft(0L)((a, x) => a + x)

  val t1 = System.nanoTime()
  println(s"BENCH_MS: ${(t1 - t0) / 1000000}")
  println(s"result=$result")
